src/main/java/mate/academy/internetshop/dao/jdbc/ProductDaoJdbcImpl.java

```diff
@@ -0,0 +1,109 @@
+package mate.academy.internetshop.dao.jdbc;
+
+import java.math.BigDecimal;
+import java.sql.Connection;
+import java.sql.PreparedStatement;
+import java.sql.ResultSet;
+import java.sql.SQLException;
+import java.sql.Statement;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import mate.academy.internetshop.dao.ProductDao;
+import mate.academy.internetshop.exceptions.DataProcessingException;
+import mate.academy.internetshop.lib.Dao;
+import mate.academy.internetshop.model.Product;
+import mate.academy.internetshop.util.ConnectionUtil;
+
+@Dao
+public class ProductDaoJdbcImpl implements ProductDao {
+    private Product getProductFromResultSet(ResultSet resultSet) throws SQLException {
```

Comment:
this method is helpful (because it is private and is using in the other methods) so it should not be on the top of class implementation. Can we move it to the bottom?

Reply:
done